Public Course Content

grantgoodmanedX opened the pull request:
## [DOC-3985](https://openedx.atlassian.net/browse/DOC-3985)

Documentation for the public course content feature, which is in Ironwood. It's based heavily on the blog post draft at https://docs.google.com/document/d/1reMV0HEhT6HZSwJinvMReBULQ_u_0orz-gmLoB0yOoU/edit#

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Subject matter expert: @pomegranited 
- [ ] Subject matter expert: @symbolist
- [ ] Doc team review (sanity check, copy edit, or dev edit?): @edx/doc
- [ ] Product review

### Testing

- [ ] Ran ./run_tests.sh without warnings or errors

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [x] Squash commits

pomegranited commented on en_us/install_operations/source/configuration/enable_public_course.rst:
```diff
@@ -0,0 +1,128 @@
+  this flag for just that course.  
+
+  .. image:: ../Images/public-add-flag.png
+         :alt: Setting a waffle flag in the Django LMS Admin.
```
The images for these two types of waffle flags are switched around.  Maybe if `../Images/public-waffle-flag.png` was named `../Images/course-waffle-flag.png` this would be easier to keep straight?

Also, since they're in a shared `Images` dir, you might want to name them according to the waffle flags we're adding here.

pomegranited commented on en_us/install_operations/source/configuration/enable_public_course.rst:
```diff
@@ -0,0 +1,128 @@
+   the other course content, will be visible to everyone.
+
+   If you change this to ``"public"``, then all of your course content, including
+   the course outline, will be visible to anyone.
```
nit: says "visible to everyone" above, but "visible to anyone" here.

pomegranited commented on en_us/shared/developing_course/controlling_content_visibility.rst:
```diff
@@ -17,6 +17,12 @@ features.
+.. only:: Open_edX
+
+  You can also configure Open edX so that courses or course outlines are visible 
+  to anyone, not just enrolled learners. For information, see 
```
nit: "not just registered and enrolled learners" ?

pomegranited commented on en_us/install_operations/source/configuration/enable_public_course.rst:
```diff
@@ -0,0 +1,128 @@
+
+.. Note::
+  Before proceeding, review :ref:`Guidelines for Updating the Open edX
+  Platform`.
```
Since these are [not server configuration changes](https://github.com/edx/edx-documentation/blob/master/en_us/install_operations/source/configuration/updating_platform.rst#L3), they're not protected by the guidelines noted here.  So I believe this note can be omitted?

grantgoodmanedX commented:
@pomegranited Thanks for your thoughtful review. I think I've addressed all of your suggestions. 

I also think that the test failure will be resolved in the build (since I'm referring to a new page that's added in the same PR).

pomegranited commented:
Thanks for getting this docs change in @grantgoodmanedX !  Can you point me to where it will be visible on the https://edx.readthedocs.io site, so I can link to it from my blog post?

grantgoodmanedX commented:
@pomegranited It's now visible at https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/enable_public_course.html
